scripts/coverage_report_jenkins.py

```diff
@@ -51,6 +118,224 @@ def get_build_numbers(client, job_name):
+
+    # Extract the scanner
+    if not ssh.run_command('unzip -d {} {}'.format(scanner_dir, scanner_zip)):
+        raise Exception(
+            'Could not extract scanner software, {}, to {}'.format(scanner_zip, scanner_dir))
+
+    # Note, all the files are underneath one directory under our scanner_dir, but we don't
+    # necessarily know the name of that directory.   Yes today, as I write this, the name
+    # will be:
+    #
+    #   sonar-scanner-$version-linux
+    #
+    # but if they decide to change its name, any code that depended on that would break.   So
+    # what will do is go into the one directory that now under our scanner_dir, and move all
+    # those files up a directory (into our scanner_dir).   tar has the --strip-components
+    # option that would have avoided this, however we are dealing with a zip file and unzip
+    # has no similar option.
+    if not ssh.run_command('cd {}; mv $(ls)/* .'.format(scanner_dir)):
+        raise Exception('Could not move scanner files into scanner dir, {}'.format(scanner_dir))
+
+    # Configure the scanner to point to the local sonarqube
+    # WARNING:  This definitely makes the assumption the only thing we need in that config is
+    #           the variable sonar.host.url set.  If that is ever wrong this will fail, perhaps
+    #           mysteriously.  So the ease of this implementation is traded off against that
+    #           possible future consequence.
+    scanner_conf = '{}/conf/sonar-scanner.properties'.format(scanner_dir)
+    if not ssh.run_command('echo "sonar.host.url={}" > {}'.format(server_url, scanner_conf)):
+        raise Exception('Could write scanner conf, {}s'.format(scanner_conf))
+
+    # Now configure the project
+    #
+    # We have sources in two directories:
+    #
+    #   - /opt/rh/cfme-gemset
+    #   - /var/www/miq/vmdb
+    #
+    # It is very important that we set sonar.sources to a comma delimited
+    # list of these directories but as relative paths, relative to /.   If
+    # we configure them as absolute paths it will only see the files /var/www/miq/vmdb.
+    # Don't know why, it just is that way.
+    #
+    # Hear is an example config:
+    #
+    #   sonar.projectKey=CFME5.9-11
+    #   sonar.projectName=CFME-11
+    #   sonar.projectVersion=5.9.0.17
+    #   sonar.language=ruby
+    #   sonar.sources=opt/rh/cfme-gemset,var/www/miq/vmdb
+    project_conf = 'sonar-project.properties'
+    local_conf = os.path.join(log_path.strpath, project_conf)
+    remote_conf = '/{}'.format(project_conf)
+    config_data = '''
+sonar.projectKey=CFME
+sonar.projectName=CFME
+sonar.projectVersion={version}
+sonar.language=ruby
+sonar.sources=opt/rh/cfme-gemset,var/www/miq/vmdb
+'''.format(
+        version=str(appliance.version).strip())
+
+    # Write the config file locally and then copy to remote.
+    print 'Writing {}'.format(local_conf)
+    f = open(local_conf, 'w')
```

Comment:
Use a context manager here,

```python
with open(local_conf, 'w') as f:
    f.write(config_data)

Reply:
Ha I discovered the thumbs up.   Life is glorious.